### PR TITLE
[FreeBSD] Unbreak libdemangle build, libbsd is not required.

### PR DESCRIPTION
**Pull request by dcci**
Trying to fix the FreeBSD build until we get proper CI :)

### CODE REVIEW

**Comment by dcci** [comment]
@swift-ci please test

File: lib/SwiftDemangle/CMakeLists.txt [hunk 1]
```diff
@@ -3,7 +3,12 @@ add_swift_library(swiftDemangle SHARED
   MangleHack.cpp
   LINK_LIBRARIES swiftDemangling)
 if(NOT APPLE)
-  target_link_libraries(swiftDemangle PRIVATE bsd)
```
Comment (jrose-apple) [inline lib/SwiftDemangle/CMakeLists.txt]: [This is all one command](https://cmake.org/cmake/help/v3.3/command/target_link_libraries.html#libraries-for-a-target-and-or-its-dependents): link the build product `swiftDemangle`, `PRIVATE`ly, to `bsd`. You can just change the condition.

Comment (dcci) [reply lib/SwiftDemangle/CMakeLists.txt]: Thank you Jordan. I still have hard time reading CMake ;)

**Comment by dcci** [comment]
Addressed Jordan's comment.

**Comment by dcci** [comment]
@swift-ci please test

**Review by jrose-apple — APPROVED** [review]

**Pull request merged**